satpy/readers/mws_l1b.py

```diff
@@ -0,0 +1,288 @@
+# Copyright (c) 2022 Pytroll Developers
+
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+"""Reader for the EPS-SG Microwave Sounder (MWS) level-1b data.
+
+Documentation: https://www.eumetsat.int/media/44139
+"""
+
+import logging
+from datetime import datetime
+
+import dask.array as da
+import numpy as np
+from netCDF4 import default_fillvals
+
+from .netcdf_utils import NetCDF4FileHandler
+
+logger = logging.getLogger(__name__)
+
+
+# dict containing all available auxiliary data parameters to be read using the index map. Keys are the
+# parameter name and values are the paths to the variable inside the netcdf
+
+AUX_DATA = {
+    'scantime_utc': 'data/navigation/mws_scantime_utc',
+    'solar_azimuth': 'data/navigation/mws_solar_azimuth_angle',
+    'solar_zenith': 'data/navigation/mws_solar_zenith_angle',
+    'satellite_azimuth': 'data/navigation/mws_satellite_azimuth_angle',
+    'satellite_zenith': 'data/navigation/mws_satellite_zenith_angle',
+    'surface_type': 'data/navigation/mws_surface_type',
+    'terrain_elevation': 'data/navigation/mws_terrain_elevation',
+    'mws_lat': 'data/navigation/mws_lat',
+    'mws_lon': 'data/navigation/mws_lon',
+}
+
+MWS_CHANNEL_NAMES_TO_NUMBER = {'1': 1, '2': 2, '3': 3, '4': 4,
+                               '5': 5, '6': 6, '7': 7, '8': 8,
+                               '9': 9, '10': 10, '11': 11, '12': 12,
+                               '13': 13, '14': 14, '15': 15, '16': 16,
+                               '17': 17, '18': 18, '19': 19, '20': 20,
+                               '21': 21, '22': 22, '23': 23, '24': 24}
```

Comment:
wouldn't be simpler to do `int(channel_name)` than using a dict like this?

Reply:
I was not sure this would actually be the final channel/band naming in the file